internal/providercache/package_install.go

```diff
@@ -26,9 +28,35 @@ import (
 // specific protocol and set of expectations.)
 var unzip = getter.ZipDecompressor{}
 
+const (
+	// httpClientRetryCountEnvName is the environment variable name used to customize
+	// the HTTP retry count for module downloads.
+	httpClientRetryCountEnvName = "TF_PROVIDER_DOWNLOAD_RETRY"
+
+	defaultRetry = 2
+)
+
+var (
+	maxRetryCount int
+)
+
+func requestLogHook(logger retryablehttp.Logger, req *http.Request, i int) {
+	if i > 0 {
+		logger.Printf("[INFO] Previous request to the provider install failed, attempting retry.")
+	}
+}
+
 func installFromHTTPURL(ctx context.Context, meta getproviders.PackageMeta, targetDir string, allowedHashes []getproviders.Hash) (*getproviders.PackageAuthenticationResult, error) {
 	url := meta.Location.String()
 
+	maxRetryCount = defaultRetry
```

Comment:
Why is maxRetryCount a global variable? It shouldn't be, or you might accidentally be modifying it from multiple threads simultaneously.

Reply:
I believe this is an artifact from copying some of the discovery retry mechanism that is in place. That uses a global variable because it uses an init() method to read the env var.

I think now that this PR is reading the env var inside the method here, we can just set it's default value inside this method and remove the global variable.